File: tests/e2e/cases/09-type-coverage/clamp_neg_i8/clamp_neg_i8_verified.cpp
/*
DEMO: Clamp-negative-to-zero, i8.

Returns 0 when x is strictly negative, x otherwise. Exercises:
  * `int8_t` parameter + return — width not seen by the existing
    integer suites (popcount_u8 covered u8, min3_i32 covered i32).
  * Signed-less-than at 8 bits (lowers to `icmp slt i8`).

Expected verdict: VERIFIED.
*/

#include <cstdint>

int8_t clamp_neg_i8(int8_t x) {
    return x < 0 ? (int8_t)0 : x;
}

File: tests/e2e/cases/09-type-coverage/clamp_neg_i8/clamp_neg_i8_verified.rs
// DEMO: clamp-negative-to-zero, i8 (Rust mirror of clamp_neg_i8_verified.cpp).
// Exercises i8 parameter + return (signed 8-bit), absent from earlier suites.

fn clamp_neg_i8(x: i8) -> i8 {
    if x < 0 { 0 } else { x }
}
